Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -160,7 +169,7 @@ public Task<bool> IsContestIpValidByContestAndIp(int contestId, string ip)
                     c.Id == contestId &&
                     (!c.IpsInContests.Any() || c.IpsInContests.Any(ai => ai.Ip.Value == ip)));
 
-        private async Task<Participant> AddNewParticipantToContest(Contest contest, bool official, string userId,
+        private async Task<Participant?> AddNewParticipantToContest(Contest contest, bool official, string userId,
```

Comment:
Either rename this AddNewParticipant..IfNotExists or remove the nullable option in the return type

Reply:
Renamed.